src/main/java/org/cirdles/topsoil/dataset/DatasetResource.java

```diff
@@ -15,22 +15,23 @@
  */
 package org.cirdles.topsoil.dataset;
 
-import java.util.List;
-
 /**
+ * Each DatasetResource handles all CRUD methods for a given Dataset
  *
- * @author John Zeringue
+ * @author parizotclement
  */
-public interface DatasetManager {
+public interface DatasetResource {
+
+    public Dataset getDataset();
```

Comment:
You should either do `getDataset`, `deleteDataset`, `saveDataset`, ... or `get`, `delete`, `save`, ...